test/e2e/shared/openstack.go

```diff
@@ -135,6 +137,132 @@ func dumpOpenStackImages(providerClient *gophercloud.ProviderClient, clientOpts
 	return nil
 }
 
+func DumpOpenStackServers(e2eCtx *E2EContext, filter servers.ListOpts) ([]servers.Server, error) {
+	providerClient, clientOpts, err := getProviderClient(e2eCtx)
+	if err != nil {
+		_, _ = fmt.Fprintf(GinkgoWriter, "error creating provider client: %s\n", err)
+		return nil, nil
+	}
+
+	computeClient, err := openstack.NewComputeV2(providerClient, gophercloud.EndpointOpts{Region: clientOpts.RegionName})
+	if err != nil {
+		return nil, fmt.Errorf("error creating compute client: %v", err)
+	}
```

Comment:
I think it's better to move the creation of the clients into a helper function.
Further, the usage of the gomage `Expect` would be better instead of the usual `err != nil` condition in the e2e ginkgo tests.